### ISSUE TITLE
Ace reporting  duplicate-id incorrectly (due to JS replacement of text)

### ISSUE THREAD

**Issue by clapierre** [daisy/ace#319]
1.1.1 Command line version

Received this error for a JavaScript replacement of text in our Javascript test in our Enhanced Image Description EPUBTest.org book.

Minor | WCAG 2.0 A | duplicate-idaXe | " Extended description using a progressively enhanced details and summary "Text/testDetailsEnhanced.xhtml#epubcfi(/4/10[extdesc-190]/6/2[js-test-src])Snippet:<b xmlns="http://www.w3.org/1999/xhtml" id="js-test-src">"JavaScript is enabled. This test passes."</b>  <b xmlns="http://www.w3.org/1999/xhtml" id="js-test-src">"JavaScript is enabled. This test passes."</b>

Here is the code for this xhtml page

```
<section id="extdesc-190" class="test">
      <h2>
        <span class="test-title">JavaScript is enabled</span>
        <br />
        <span class="test-id">extdesc-190</span>
      </h2>
      <p class="desc">
        This test passes if the following condition is met when JavaScript is
        enabled:
      </p>
      <p>
        You see the text &quot;<b id="js-test-src">JavaScript is enabled. This test passes.</b>&quot; between
        the following two horizontal lines
      </p>

      <hr />
      <div id="js-test-dest">JavaScript is disabled. This test fails.</div>
      <hr />
      <p>
        <b>Note</b>: this test is included because subsequent tests use
        JavaScript to add accessibility affordances. If this test fails, other
        tests on this page may not work as expected.
      </p>
    </section>

```

    <script>
      document.querySelectorAll(".overlaid").forEach(EnhancedDetails.enhance);
      // make extdesc-190 pass
      document.getElementById(
        "js-test-dest"
      ).innerHTML = document.getElementById("js-test-src").outerHTML;
    </script>

Here is a[ link to the testBook](https://benetech.box.com/s/0vxltp0k4acpg3a58bteyewuupt4lm9w), this is test # 190

**Comment by marisademeglio** [comment]
Ace performs its checks on the version of the document that a browser would see, via a "headless browser". So I believe the error is correct, because `id="js-test-src"` would appear on two elements after  the script runs . I can think of a few workarounds:

1. Use `<noscript>` tags
```
<p>Javascript <noscript>is not</noscript> enabled.</p>
```
2. Rather than copy the text from an element, just type it out. 
3. Wrap the bold text in another element with an id, and copy that element's innerHTML to the innerHTML of `#js-test-dest` so that no IDs are duplicated.
 
If for some reason Ace still reports an error, then please let us know.

**Comment by danielweck** [comment]
Closing this issue as the behaviour is expected.
